src/mgr/Mgr.h

```diff
@@ -19,6 +19,7 @@
 
 #include "mds/FSMap.h"
 #include "messages/MFSMap.h"
+#include "messages/MMonMap.h"
```

Comment:
no need to `#include "messages/MMonMap.h"`, and we don't even need to forward declare `MMonMap`, what we need is a method `handle_mon_map()` which is called whenever a monmap is received. and monc will decode it for us.

Reply:
i see. it's a little bit awkward. let me note down the problem in my own words:

assuming we have a cluster with mon.a and mon.b. and they have formed a quorum where mon.a is the leader. if we send "mon metadata" to monitor in `Mgr::handle_mon_map()`, we will have:

1. add mon.c to this cluster
2. mon.c sends a `MMonJoin` to mon.a for joining the cluster.
3. mon.a updates the monmap to include the new joiner
4. mgr.x is updated with the latest monmap
5. mgr.x tries to request for the metadata of mon.c, but since the monitors have not finished the election. the metadata are not updated yet. so mgr.x get a `-EINVAL`, see `Monitor::get_mon_metadata()
6. mon.c manages to join the quorum, and the metadata gets updated. see `Monitor::win_election()`. and monmap is not updated for the completion of election. and mgr.x won't ask the cluster for the updated metadata
6. mgr.x

Reply:
again, no need to include `MMonMap.h`.